tests/currency-utils/test_currency_tools.py

```diff
@@ -97,3 +97,30 @@ def test_invalid_to_wei_values(value, unit):
 
     with pytest.raises(ValueError):
         from_wei(value, unit)
+
+
+@pytest.mark.parametrize(
```

Comment:
This would be best split into two separate tests without the need for the conditional.

Conditionals make it pretty hard to debug the test itself. Instead, I think adding all the expected fail cases in one test and the expected `assert` cases in another is easier to read, debug, and to know that each test is going through the same logic and doing what we expect in every case.